internal/services/v1/hash.go

```diff
@@ -0,0 +1,58 @@
+package v1
+
+import (
+	"strconv"
+
+	"github.com/golang/protobuf/ptypes/struct"
+
+	"github.com/authzed/authzed-go/proto/authzed/api/v1"
+
+	"github.com/authzed/spicedb/pkg/spiceerrors"
+)
+
+func computeCallHash(apiName string, consistency *v1.Consistency, arguments map[string]any) (string, error) {
```

Comment:
perhaps testing some basic permutations, to make sure the hash is different for different requests, will help.
Please also note that ProtoJSON is not stable and it's going to make 2 equal requests fail. You had code around to handle this with caveats, so you could probably refactor it

Reply:
Moved to using shared code and added tests